pkg/bindings/images/build.go

```diff
@@ -607,6 +607,75 @@ func prepareContainerFiles(containerFiles []string, contextDir string, options *
 	return &out, nil
 }
 
+// prepareLocalContainerFiles prepares container files for local build by translating paths for remote system and creating temporary files.
+// WARNING: Caller must ensure tempManager.Cleanup() is called to remove any temporary files created.
+func prepareLocalContainerFiles(containerFiles []string, contextDir string, tempManager *remote_build_helpers.TempFileManager, translationLocalAPIMap map[string]string) (*BuildFilePaths, error) {
+	out := BuildFilePaths{
+		tarContent:        []string{contextDir},
+		newContainerFiles: []string{}, // dockerfile paths, relative to context dir, ToSlash()ed
+	}
+
+	for _, c := range containerFiles {
+		// Don not add path to containerfile if it is a URL
+		if strings.HasPrefix(c, "http://") || strings.HasPrefix(c, "https://") {
+			out.newContainerFiles = append(out.newContainerFiles, c)
+			continue
+		}
+		isStdIn := c == "/dev/stdin"
+		if isStdIn {
+			stdinFile, err := tempManager.CreateTempFileFromReader(contextDir, "podman-build-stdin-*", os.Stdin)
+			if err != nil {
+				return nil, fmt.Errorf("processing containerfile from stdin: %w", err)
+			}
+			c = stdinFile
+		}
+		c = filepath.Clean(c)
+		cfDir := filepath.Dir(c)
+		if absDir, err := filepath.EvalSymlinks(cfDir); err == nil {
+			name := filepath.ToSlash(strings.TrimPrefix(c, cfDir+string(filepath.Separator)))
+			c = filepath.Join(absDir, name)
+		}
+
+		containerfile, err := filepath.Abs(c)
+		if err != nil {
+			logrus.Errorf("Cannot find absolute path of %v: %v", c, err)
+			return nil, err
+		}
+
+		// If Containerfile does not exist, assume it is in context directory and do Not add to tarfile
+		isInContextDir := false
+		if err := fileutils.Lexists(containerfile); err != nil {
+			if !errors.Is(err, fs.ErrNotExist) {
+				return nil, err
+			}
+			containerfile = c
+			isInContextDir = true
+		}
+
+		remoteContainerfile := containerfile
+		if isStdIn {
+			remoteContainerfile, err = specgen.ConvertWinMountPath(remoteContainerfile)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			var ok bool
+			remoteContainerfile, ok = translationLocalAPIMap[containerfile]
+			if !ok {
+				if !isInContextDir {
+					return nil, fmt.Errorf("containerfile %q not found in translation map", containerfile)
```

Comment:
```suggestion
					return nil, fmt.Errorf("Containerfile %q not found in translation map", containerfile)
```
I'll just add this one and it's a nit, but a big nit.  Customer facing verbaiage should list this with a capital `C`.  I'll just comment here, but there are other intances.